vscode/src/edit/edit-manager.ts

```
@@ -306,9 +307,23 @@ export class EditManager implements vscode.Disposable {
         }
 
         this.logExecutedTaskEvent(task)
-        this.options.fixupController.startDecorator(task)
-        const provider = this.getProviderForTask(task)
-        await provider.startStreamingEdit()
+        try {
+            this.options.fixupController.startDecorator(task)
+            const provider = this.getProviderForTask(task)
+            await provider.startStreamingEdit()
+        } catch (error: unknown) {
+            const errorObject =
+                error instanceof Error
+                    ? error
+                    : new Error(typeof error === 'string' ? error : String(error))
+            // If there's an error, abort the task by sending an error event
+            this.options.fixupController.error(task.id, errorObject)
+
+            logError('startStreamingEditTask error:', errorObject.message)
```

Why do we need special handling here now ? Won't error always propogate to upstream. Did you observe any instances where errors are thrown ? I am assuming there would be upstream services handling the errors, so  can this `vscode.window.showErrorMessage ` lead to duplication.
Adding @umpox who has more idea about the application code.

---

@hitesh-1997 not in the case of the `startStreamingEditTask` task. When I started with this ticket there was a logic bug within `trimEmptyLinesFromRange` https://github.com/sourcegraph/cody/blob/ichimdav/apply_markdown/vscode/src/non-stop/decorations/compute-decorations.ts#L82

Which gave the error that you see in the comment I left here
https://github.com/sourcegraph/cody/pull/7439#issuecomment-2785509278

Before my code it would simply fail silently but the user never new and the smart apply was just spinning without it triggering an error that allows you to see the error.

I have mentioned in the comment above that it might be excessive to add the vscode `showErrorMessage` which is why I was asking if we should have it or not.